### ISSUE TITLE
Validation Icon - Missing

### ISSUE THREAD

**Issue by ejio** [formvalidation/support#126]
I've just noticed that the validation icon fails to appear on the initial field, however the help text (validation messages) still appears and the icons appear on all other fields.

    <i class="form-control-feedback fv-icon-no-label" data-fv-icon-for="email" style=""></i>

is the HTML on validation, in comparison to other fields where this is displayed (correctly)

    <i class="form-control-feedback fv-icon-no-label glyphicon glyphicon-remove colora94442" data-fv-icon-for="email_confirm" style=""></i>

![screen shot 2015-04-23 at 15 27 25](https://cloud.githubusercontent.com/assets/12000599/7299358/49c2adbc-e9cd-11e4-81e5-906c73e816f2.png)

HTML

    <div class="input-group">
                        <span class="input-group-addon"><i class="fa fa-fw fa-envelope"></i></span>
                        <input id="email" name="email" type="email" class="form-control" value="" placeholder="email" required="required">
                    </div>

Validator

        email: {
                enabled: true,
                verbose: false,
                validators: {
                    notEmpty: {
                        message: 'field is required'
                    },
                    emailAddress: {
                        message: 'field format is invalid'
                    },
                    regexp: {
                        regexp: '^[^@\\s]+@([^@\\s]+\\.)+[^@\\s]+$',
                        message: 'field input is invalid'
                    },
                    remote: {
                        type: 'POST',
                        url: "http://" + servername + "/path?request=email&type=json",
                        crossDomain: false,
                        name: 'data',
                        data: {
                            data: 'email',
                        },
                        dataType: 'json',
                        validKey: 'valid',
                        message: 'not allowed - may be registered or restricted'
                    }
                }
            },

**Comment by nghuuphuoc** [comment]
Can you please give me the full HTML of elements that contain email, confirm_email and password fields?

**Comment by ejio** [comment]
Sure.

    <form id="join" class="form-horizontal" role="form" method="post" action="#">
                    <div class="input-group">
                        <span class="input-group-addon"><i class="fa fa-fw fa-envelope"></i></span>
                        <input id="email" name="email" type="email" class="form-control" value="" placeholder="email" required="required">
                    </div>
                    <div class="input-group">
                        <span class="input-group-addon"><i class="fa fa-fw fa-question"></i></span>
                        <input id="email_confirm" name="email_confirm" type="email" class="form-control" value="" placeholder="confirm email" required="required">
                    </div>
                    <div class="input-group">
                        <span class="input-group-addon"><i class="fa fa-fw fa-lock"></i></span>
                        <input id="password" name="password" type="password" class="form-control" value="" placeholder="password" required="required">
                    </div>
                    <div class="progress password-meter" id="password-meter">
                        <div class="progress-bar"></div>
                    </div>
                    <div class="form-group">
                        <div class="col-sm-12 controls">
                            <div id="validationmessages"></div>
                            <button type="submit" id="submitButton" name="submitButton" class="btn">Join</button>
                        </div>
                    </div>
                </form>

**Comment by ejio** [comment]
I've rounded this down a little, it seems to be in relation to the verbose and remote validator.

I'll feedback when I have more.

**Comment by ejio** [comment]
Here's what I've found:

- removing 'remote' validator from 'email' = no icon
- removing 'verbose' from 'email' = icon <- the problem?

This behaviour seems to only apply to 'email', as on 'password', where I have both 'remote' and 'verbose' the icon appears as expected.

**Comment by nghuuphuoc** [comment]
It's not caused by the `verbose` option. 

There are few things you should care about:

1) By default, in order to place the icon and messages, FormValidation will look for the parent container of field which usually has `.form-group` class.
If the field isn't placed inside a `.form-group` element, you need to indicate its container it via the [row](http://formvalidation.io/settings/#field-row) option.

This option is mostly used in a [complex form](http://formvalidation.io/examples/complex-form/).
In your case, we need to set row option as `.input-group`.

2) Your remote validator options have some one with default values. You don't need it:

```javascript
               remote: {
                    type: 'POST',
                    url: "http://" + servername + "/path?request=email&type=json",
                    crossDomain: false,
                    name: 'data',
                    data: {
                        data: 'email',
                    },
                    dataType: 'json',   // Don't need it
                    validKey: 'valid',    // Don't need it
                    message: 'not allowed - may be registered or restricted'
                }
```

Here is the fiddle: http://jsfiddle.net/nghuuphuoc/qd970ttw/3/

The result: 

![icons](https://cloud.githubusercontent.com/assets/4226036/7301996/773fd696-ea10-11e4-8815-0203481ca873.png)

**Comment by ejio** [comment]
I've implemented your changes, but am still seeing some some odd behaviour (of course, it could just be me...), here goes:

These two jsfiddles are exactly the same in content, barring the initial field name being validated.

http://jsfiddle.net/amkLky0e/2/ - initial field name is email
http://jsfiddle.net/n1o7ff6e/2/ - initial field name is test

The validation display is different.

email and email_confirm = no icon
test and email_confirm = icon

In the source posted here I modified the field names for clarify, however they were originally join_email, join_email_confirm, join_password. Is it possible that join_email and join_email_confirm are conflicting in some fashion due to the common name they begin with 'join_email'?

Hopefully this all makes some sense?
